grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -256,8 +257,9 @@ public void start(Listener<O> responseListener, Metadata metadata) {
                     }
                     return HttpResponse.ofFailure(status.asRuntimeException());
                 };
-        final HttpResponse res = initContextAndExecuteWithFallback(
-                httpClient, ctx, endpointGroup, HttpResponse::of, errorResponseFactory);
+        final HttpPreClient execution =
+                preprocessors.decorate(TailPreClient.of(client, HttpResponse::of, errorResponseFactory));
```

Comment:
Suggestion) Should you initiate `HttpPreClient` in the constructor of `ArmeriaChannel` and reuse it? It seems possible if `credentials == null`.

This PR is an initial step in the preprocessor implementation, so you can consider micro-optimization later.